staging/src/k8s.io/apiserver/pkg/cel/environment/base.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package environment
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/google/cel-go/cel"
+	"github.com/google/cel-go/ext"
+
+	celconfig "k8s.io/apiserver/pkg/apis/cel"
+	"k8s.io/apiserver/pkg/cel/library"
+)
+
+// WARNING: Do not change to supportedEnvOpts or supportedProgramOpts
+// without first applying the change to storageEnvOpts or storageProgramOpts
+// for at least one Kubernetes minor release.
+// All changes must be backward compatible.
+
+var supportedEnvOpts = []cel.EnvOption{
+	cel.HomogeneousAggregateLiterals(),
+	// Validate function declarations once during base env initialization,
+	// so they don't need to be evaluated each time a CEL rule is compiled.
+	// This is a relatively expensive operation.
+	cel.EagerlyValidateDeclarations(true),
+	cel.DefaultUTCTimeZone(true),
+}
+
+var supportedProgramOpts = []cel.ProgramOption{
+	cel.EvalOptions(cel.OptOptimize, cel.OptTrackCost),
+	cel.CostLimit(celconfig.PerCallLimit),
+}
+
+var supportedLibs = []cel.EnvOption{
+	ext.Strings(),
+	library.URLs(),
+	library.Regex(),
+	library.Lists(),
+	library.Authz(),
+}
+
+// Modify these variables when adding language configuration or libraries that are planned to
+// made available in a future Kubernetes release.
+var storageOnlyEnvOpts []cel.EnvOption
+var storageOnlyProgramOpts []cel.ProgramOption
+var storageOnlyLibs []cel.EnvOption
+
+// BaseEnvSet provides the common CEL base environments for Kubernetes.
+//
+// This environment contains no CEL variable definitions and should be extended
+// to construct environments with the appropriate variable definitions.
+func BaseEnvSet() *EnvSet {
```

Comment:
this accessor may need to take the major/minor we want to be compatible with (and partition the envs we create per major/minor)

Reply:
SGTM. I'll place a default value in as the major/minor for now and we can wire in what we need later for skip upgrades.